src/atdf/field.rs

```diff
@@ -6,10 +6,15 @@ use crate::ElementExt;
 pub fn parse(
     bitfield_el: &xmltree::Element,
     value_groups: &atdf::values::ValueGroups,
+    mode_name: Option<String>,
```

Comment:
If you pass the `mode_name` as an `Option<&str>`, you don't need to clone it in the caller.  This is fine because you only use the `mode_name` to copy it into the field name using `format!()`.

(Performance wise, this doesn't matter here, but it's good style to not do unnecessary clones)